modules/internal/ChapelRange.chpl

```diff
@@ -144,7 +144,8 @@ module ChapelRange {
   
     var _low: idxType = 1;                         // lower bound
     var _high: idxType = 0;                        // upper bound
-    var _stride: strType = 1;                      // signed stride of range
+    var _stride: if stridable then strType else void =
+                   if stridable then 1 else _void; // signed stride of range
```

Comment:
I think that would work, but I'm not sure if that's better than having a named void value or a built-in function that generates void values...  I worry that "0:void" won't be very meaningful to most readers (or we could do things like "100432:void" just to be even more confusing.  :)